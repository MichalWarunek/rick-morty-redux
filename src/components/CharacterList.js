import React from 'react';
import { connect } from 'react-redux';
import selectCharacters from '../selectors/characters';
import CharacterListItem from './CharacterListItem';
import { Container, Row, Col } from 'reactstrap';
import Pages from './Pages';
import { Link } from 'react-router-dom';


const CharacterList = (props) => (
  <div>
  <Container>
    <Row>
      {
        props.characters.length === 0 ? (
          <p className="form__error">No characters</p>
        ) : (
          props.characters.map((character) => {
            return <CharacterListItem key={character.id} {...character} />;
          }
          
          )
        )
        
      }
      
      <Col md="2" sm="4" xs="6">
        <div className="add-character-button">
          <Link to={`/create`}>
            <img className="add-image" src="/images/add.jpg" 
            />
          </Link>
        </div>
      </Col>
      

    </Row>
    <Pages />
  </Container>
  </div>
);





const mapStateToProps = (state) => {
  return {
    // characters: state.characters
    characters: selectCharacters(state.characters, state.filters)
  };
};


export default connect(mapStateToProps)(CharacterList);