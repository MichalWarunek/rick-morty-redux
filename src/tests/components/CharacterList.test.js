import React from 'react';
import { shallow } from 'enzyme';
import CharacterList from '../../components/CharacterList';

test('should correctly render character list', () => {
  const wrapper = shallow(<CharacterList />);
  expect(wrapper).toMatchSnapshot();
});