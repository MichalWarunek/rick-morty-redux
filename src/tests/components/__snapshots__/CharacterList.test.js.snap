// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render character list 1`] = `
<div>
  This is CharacterList
  <CharacterListItem />
</div>
`;
